tests/Unit/NumericalAlgorithms/Spectral/Test_Projection.cpp

```diff
@@ -0,0 +1,260 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <cstddef>
+#include <numeric>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Matrix.hpp"
+#include "ErrorHandling/Assert.hpp"
+#include "NumericalAlgorithms/Spectral/LegendreGaussLobatto.hpp"
+#include "NumericalAlgorithms/Spectral/Projection.hpp"
+
+namespace {
+DataVector apply_matrix(const Matrix& m, const DataVector& v) noexcept {
```

Comment:
Should we get `Matrix` and `DataVector` algebra wrappers in first, with #666? Same with your `inner_product` below.

Reply:
Yeah, it would be nice if we could use those here.  There might be a few places above as well, I'd have to check.

Reply:
`inner_product` should be covered by the `dot_product` added in #673, if we decide to merge that.

Reply:
I think all the Matrix changes have finished now, but I don't see the functionality I was waiting on here.